src/openrct2/Game.cpp

```diff
@@ -359,7 +359,7 @@ void game_fix_save_vars()
     // Fix possibly invalid field values
     for (auto peep : EntityList<Guest>())
     {
-        if (peep->CurrentRideStation >= MAX_STATIONS)
+        if (peep->CurrentRideStation.ToUnderlying() >= MAX_STATIONS)
```

Comment:
There are a couple more instances where the station index is checked with `>= MAX_STATIONS`, but if I understand correctly this can only evaluate to true when the station index is `STATION_INDEX_NULL` (255). Higher values would overflow back to 0, since `StationIndex` was of type `uint8_t`.